test/services/database/post_repository_test.dart

```diff
@@ -47,6 +54,43 @@ void main() {
       expect(actualPost.exists, false);
     });
 
+    test("Delete deletes subcollections", () async {
+      await setPostFirestore(post, firestore);
+      final postRef =
+          firestore.collection(PostFirestore.collectionName).doc(post.id.value);
+      // Check that the post is in the db
+      final dbPost = await postRef.get();
+      expect(dbPost.exists, true);
+
+      final commentRepository = CommentRepositoryService(firestore: firestore);
+      await CommentFirestoreGenerator()
+          .addComments(4, post.id, commentRepository);
+
+      final upvoteRepository = UpvoteRepositoryService<PostIdFirestore>(
+        firestore: firestore,
+        parentCollection: firestore.collection(PostFirestore.collectionName),
+        voteScoreField: VoteFirestore.votersSubCollectionName,
+      );
```

Comment:
I think a `PostUpvoteRepsoitoryService` class would also be interesting. See my previous comment for all the details of my reasoning.

Reply:
See above discussion.